server/src/test/java/org/elasticsearch/cluster/metadata/ViewMetadataTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractChunkedSerializingTestCase;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.elasticsearch.cluster.metadata.ViewTestsUtils.randomName;
+import static org.elasticsearch.cluster.metadata.ViewTestsUtils.randomView;
+
+public class ViewMetadataTests extends AbstractChunkedSerializingTestCase<ViewMetadata> {
+
+    @Override
+    protected ViewMetadata doParseInstance(XContentParser parser) throws IOException {
+        return ViewMetadata.fromXContent(parser);
+    }
+
+    @Override
+    protected ViewMetadata createTestInstance() {
+        return randomViewMetadata();
+    }
+
+    @Override
+    protected ViewMetadata mutateInstance(ViewMetadata instance) {
+        Map<String, View> views = new HashMap<>(instance.views());
+        views.replaceAll((name, view) -> randomView(name));
+        return new ViewMetadata(views);
+    }
+
+    @Override
+    protected ViewMetadata createXContextTestInstance(XContentType xContentType) {
+        return randomViewMetadata();
+    }
```

Comment:
Nit: no need to override this method; that's already the default: https://github.com/elastic/elasticsearch/blob/c6f47cc2184edbbb4d6869beacc8459a1624838b/test/framework/src/main/java/org/elasticsearch/test/AbstractChunkedSerializingTestCase.java#L37-L40
```suggestion
```